holidays/holiday_base.py

```diff
@@ -427,95 +426,110 @@ def __getattr__(self, name):
         except AttributeError as e:
             # This part is responsible for _add_holiday_* syntactic sugar support.
             add_holiday_prefix = "_add_holiday_"
-            # Raise early if prefix doesn't match to avoid regex checks.
+            # Raise early if prefix doesn't match to avoid patterns checks.
             if name[: len(add_holiday_prefix)] != add_holiday_prefix:
                 raise e
 
+            tokens = name.split("_")
+
             # Handle <month> <day> patterns (e.g., _add_holiday_jun_15()).
-            month_day = re.match(r"_add_holiday_(\w{3})_(\d{1,2})", name)
-            if month_day:
-                month, day = month_day.groups()
-                return lambda name: self._add_holiday(
-                    name,
-                    date(self._year, getattr(gregorian, month.upper()), int(day)),
-                )
+            if len(tokens) == 5:
+                month, day = tokens[3:5]
+                if len(month) == 3 and month.isalpha() and len(day) < 3 and day.isdigit():
```

Comment:
```suggestion
                *_, month, day = tokens
                if month in MONTHS and day in DAYS:
```


And `calendars::gregorian` should contain 

```
DAYS = set(str(d) for d in range(1, 32))
MONTHS = set(
    m.lower()
    for m in ("JAN", "FEB", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV", "DEC")
)
WEEKDAYS = set(wd.lower() for wd in ("MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"))

```

Reply:
I tried to implement this. I made `MONTHS` and `WEEKDAYS` as mappings, it should be faster to get numbers than `getattr(gregorian, ...)`.